components/sbm-recipes-mule-to-boot/src/main/java/org/springframework/sbm/mule/MigrateMuleToBoot.java

```diff
@@ -30,23 +31,34 @@
 import org.springframework.sbm.java.migration.conditions.HasTypeAnnotation;
 import org.springframework.sbm.mule.actions.JavaDSLAction2;
 import org.springframework.sbm.mule.conditions.MuleConfigFileExist;
+import org.springframework.sbm.project.resource.SbmApplicationProperties;
 
 import java.util.List;
 
+@RequiredArgsConstructor
 @Configuration
 public class MigrateMuleToBoot {
+    private final SbmApplicationProperties sbmProperties;
 
     @Autowired
     private JavaDSLAction2 javaDSLAction2;
 
-
     @Bean
     public Recipe muleRecipe() {
+        String name = "migrate-mule-to-boot";
+        String description = "Migrate Mulesoft 3.9 to Spring Boot.";
+
+        // Flag to enable TriggerMesh ransformation mode
+        if (sbmProperties.isMuleTriggerMeshTransformEnabled()) {
```

Comment:
We should better create  two bean methods here, calling one factory method that creates the recipe twice with the flag true and false. Taht would then finally allow to remove the application property :)